exporters/stats_stackdriver/src/test/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterTest.java

```diff
@@ -0,0 +1,208 @@
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    doReturn(mockCreateMetricDescriptorCallable).when(mockStub).createMetricDescriptorCallable();
+    doReturn(mockCreateTimeSeriesCallable).when(mockStub).createTimeSeriesCallable();
+    doReturn(null).when(mockCreateMetricDescriptorCallable)
+        .call(any(CreateMetricDescriptorRequest.class));
+    doReturn(null).when(mockCreateTimeSeriesCallable)
+        .call(any(CreateTimeSeriesRequest.class));
+  }
+
+  @Test
+  public void createWithNullCredentials() throws IOException {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("credentials");
+    StackdriverExporter.createWithCredentialsAndProjectId(null, PROJECT_ID, ONE_SECOND);
+  }
+
+  @Test
+  public void createWithNullProjectId() throws IOException {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("projectId");
+    StackdriverExporter.createWithCredentialsAndProjectId(
+        GoogleCredentials.newBuilder().build(), null, ONE_SECOND);
+  }
+
+  @Test
+  public void createWithNullDuration() throws IOException {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("exportInterval");
+    StackdriverExporter.createWithCredentialsAndProjectId(
+        GoogleCredentials.newBuilder().build(), PROJECT_ID, null);
+  }
+
+  @Test
+  public void createWithNegativeDuration() throws IOException {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("Duration must be positive");
+    StackdriverExporter.createWithCredentialsAndProjectId(
+        GoogleCredentials.newBuilder().build(), PROJECT_ID, Duration.create(-1, 0));
+  }
+
+  @Test
+  public void createHandlerTwice() throws IOException {
+    StackdriverExporter.createWithCredentialsAndProjectId(
+        GoogleCredentials.newBuilder().build(), PROJECT_ID, ONE_SECOND);
+    thrown.expect(IllegalStateException.class);
+    thrown.expectMessage("Stackdriver exporter is already created.");
+    StackdriverExporter.createWithCredentialsAndProjectId(
+        GoogleCredentials.newBuilder().build(), PROJECT_ID, ONE_SECOND);
+  }
+
+  @Test
+  public void registerViewAndExport() throws IOException {
+    View view = View.create(
+        VIEW_NAME, VIEW_DESCRIPTION, MEASURE, SUM, Arrays.asList(KEY), CUMULATIVE);
+    ViewData viewData = ViewData.create(
+        view,
+        ImmutableMap.of(Arrays.asList(VALUE), SumDataLong.create(1)),
+        CumulativeData.create(Timestamp.fromMillis(100), Timestamp.fromMillis(200)));
+    doNothing().when(mockViewManager).registerView(view);
+    doReturn(viewData).when(mockViewManager).getView(VIEW_NAME);
+
+    StackdriverExporter exporter = new StackdriverExporter(
+        PROJECT_ID, new FakeMetricServiceClient(mockStub), ONE_SECOND, mockViewManager);
+    exporter.registerView(view);
+
+    verify(mockStub, times(1)).createMetricDescriptorCallable();
+    verify(mockStub, timeout(1000).times(1)).createTimeSeriesCallable();
```

Comment:
Does the timeout need to match the export interval for this test to pass?  It would help to have a comment explaining the timing in this test.